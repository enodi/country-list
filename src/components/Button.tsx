import React from "react";

type ButtonProps = {
    title: string;
    handleClick?: () => void;
};

const Button: React.FunctionComponent<ButtonProps> = ({
    title,
    handleClick,
}) => (
    <button onClick={handleClick} className="button">
        {title}
    </button>
);

export default Button;
