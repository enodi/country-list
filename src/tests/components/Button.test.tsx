import React from "react";
import { shallow } from "enzyme";

import Button from "../../components/Button";

describe("Button Component", () => {
    let wrapper: any, handleClick: () => void;
    beforeEach(() => {
        handleClick = jest.fn();
        wrapper = shallow(<Button title="Search" handleClick={handleClick} />);
    });

    it("should render Button component correctly", () => {
        expect(wrapper).toMatchSnapshot();
    });

    test("should call handleClick on button click", () => {
        wrapper.find("button").simulate("click");
        expect(handleClick).toHaveBeenCalled();
    });
});
