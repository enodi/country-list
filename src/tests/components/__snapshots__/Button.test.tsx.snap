// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component should render Button component correctly 1`] = `
<button
  className="button "
  disabled={false}
  onClick={[MockFunction]}
>
  Search
</button>
`;
